_posts/2016-07-09-making-dnssec-work-for-an-aws-hosted-saas.markdown

```diff
@@ -0,0 +1,190 @@
+---
+title: Making DNSSEC Work for an AWS-Hosted SaaS
+layout: blog_post
+date: 2016-07-09
+description: Not everything in the Internet clicks together as easily as you might think. This post details the approach that we took to solve the hard problem of securing DNS for our government agency's SaaS servers - as well as their customers' CNAMEs to the SaaS servers - in the AWS environment.
+author: Nick Marden
+---
+
+### Introduction
+
+Software-as-a-Service is a great thing. With just a swipe of a credit card, what once took months to provision and set up has now become something that can be purchased and configured before lunch.  But as great as it is to use a SaaS service, building a reliable one can be equally challenging - especially in the cloud, where servers can disappear at any moment and infrastructure is expected to heal dynamically.
+
+[search.usa.gov](http://search.digitalgov.gov/) is a SaaS that serves thousands of other government agencies. During our recent project to migrate the search.usa.gov server infrastructure to Amazon Web Services' (AWS) Elastic Compute Cloud (EC2), we were presented with a challenging combination of requirements whose solution seems worth sharing:
```

Comment:
In this case I wanted to introduce the concept of the search.usa.gov SaaS, whose documentation (strangely) is visible at http://search.digitalgov.gov rather than http://search.usa.gov. Later on in the article, I am referring to the hostname `search.usa.gov` specifically, and not the product. That's the distinction I was trying to make anyway.